File: unibakonto/src/main/java/sk/pluk64/unibakonto/IsUnibaKonto.java
package sk.pluk64.unibakonto;

import java.util.List;
import java.util.Map;

public interface IsUnibaKonto {
    String getUsername();

    String getPassword();

    void login() throws Util.ConnectionFailedException;

    Boolean isLoggedIn(boolean refresh);

    Boolean isLoggedIn();

    Map<String, Balance> getBalances() throws Util.ConnectionFailedException;

    String getClientName() throws Util.ConnectionFailedException;

    List<Transaction> getAllTransactions() throws Util.ConnectionFailedException;

    List<Transaction> getTransactions() throws Util.ConnectionFailedException;

    List<CardInfo> getCards() throws Util.ConnectionFailedException;
}

File: unibakonto/src/main/java/sk/pluk64/unibakonto/IsUnibaKonto.kt
package sk.pluk64.unibakonto

interface IsUnibaKonto {
    val username: String

    val password: String

    val isLoggedIn: Boolean

    val balances: Balances
        @Throws(Util.ConnectionFailedException::class)
        get() = TODO()

    val clientName: String
        @Throws(Util.ConnectionFailedException::class)
        get() = TODO()

    val allTransactions: List<Transaction>
        @Throws(Util.ConnectionFailedException::class)
        get() = TODO()

    val transactions: List<Transaction>
        @Throws(Util.ConnectionFailedException::class)
        get() = TODO()

    val cards: List<CardInfo>
        @Throws(Util.ConnectionFailedException::class)
        get() = TODO()

    @Throws(Util.ConnectionFailedException::class)
    fun login()

    fun isLoggedIn(refresh: Boolean): Boolean
}
